### PR TITLE
Revert "images/cilium-test: New test suite image"

### PR DESCRIPTION
**Pull request by pchaigno**
This reverts pull requests https://github.com/cilium/cilium/pull/12941 and https://github.com/cilium/cilium/pull/15605.

This code isn't used anywhere anymore.

cc @errordeveloper

### CODE REVIEW

File: .github/workflows/build-images-ci.yaml [hunk 1]
```diff
@@ -311,19 +311,6 @@ jobs:
             cp /tmp/docker-cache-${{ matrix.name }}/tmp/go-pkg-cache.tar.gz /tmp/.cache/${{ matrix.name }}/
           fi
 
-  # we need to build cilium-test separately
```
Comment (aanm) [inline .github/workflows/build-images-ci.yaml:314]: All of this until line 438 can be deleted

**Review by aanm — CHANGES_REQUESTED** [review]

**Review by nebril — APPROVED** [review]

File: .github/workflows/build-images-ci.yaml [hunk 2]
```diff
@@ -311,131 +311,6 @@ jobs:
...
-          if [ -f /tmp/docker-cache-${{ matrix.name }}/tmp/go-build-cache.tar.gz ]; then
-            cp /tmp/docker-cache-${{ matrix.name }}/tmp/go-build-cache.tar.gz /tmp/.cache/${{ matrix.name }}/
-          fi
-          if [ -f /tmp/docker-cache-${{ matrix.name }}/tmp/go-pkg-cache.tar.gz ]; then
-            cp /tmp/docker-cache-${{ matrix.name }}/tmp/go-pkg-cache.tar.gz /tmp/.cache/${{ matrix.name }}/
-          fi
-
   image-digests:
     if: ${{ always() &&
       (needs.build-and-push-with-qemu.result == 'success' || needs.build-and-push-with-qemu.result == 'skipped') }}
```
Comment (aanm) [inline .github/workflows/build-images-ci.yaml]: this `build-and-push-with-qemu` reference needs to removed

**Review by aanm — CHANGES_REQUESTED** [review]

**Review by aanm — APPROVED** [review]

**Comment by pchaigno** [comment]
k8s-1.16-kernel-4.9 failed with known flake https://github.com/cilium/cilium/issues/22404. That flake is unrelated to the present changes. Reviews are in. Merging.

**Comment by pchaigno** [comment]
Marking this for backports to v1.13 as we keep getting dependency updates for that directory there.

**Pull request merged**